design-docs/improved-console.md

```diff
@@ -0,0 +1,283 @@
+# Improved Console Output
+
+## Primary Goals
+* Improve perceived performance by making user aware of background work
+* Modern feel
+* User extensibility
+
+### User Goals for Console
+In priority order, the user wants to answer the following questions:
```

Comment:
I'd think different users have different priorities. Personally, I'd be mostly interested in "what Gradle is going now" as top priority.